### PR TITLE
Tgc 344 consent page

### PR DESCRIPTION
**Pull request by Andrew-Folga**
This PR introduces following changes:

- introduces consent-confirmation nunjucks template for consent and confirmation pages
- provides implementation for consent question with relevant answers and designs
- tracks user answer
- Improves narrow tests with interaction tests

Note: This PR does not repopulate previous answers when traversing pages back and forth yet.

Screenshot 2024-11-11 at 22 53 12

### CODE REVIEW

File: src/config/machines/example-grant-machine.js [hunk 1]
```diff
@@ -155,38 +156,57 @@ export const exampleGrantMachine = createMachine({
...
       },
       meta: {
-        currentPageId: 'second-question', // NOSONAR:S1192 - need to improve this later
+        templateId: 'consent-confirmation',
+        currentPageId: 'consent', // NOSONAR:S1192 - need to improve this later
         previousPageId: 'country',
-        nextPageId: 'final',
+        nextPageId: 'confirmation',
+        messageHeader: 'Confirm and send',
+        messageContent: `I confirm that, to the best of my knowledge, the details I have provided are correct.</br></br>
```
Comment (swdpcomputing) [inline src/config/machines/example-grant-machine.js]: Inline definition of html within config was one of the markers that made Cap's code feel bloated. Do we want to be doing that here? Happy if we have plans to refactor later.

File: src/views/pages/consent-confirmation.njk [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        <hr/>
+        <p>
+          The Department for Environment, Food and Rural Affairs (Defra) is the data controller for personal data you give to RPA. For information on how we handle personal data go to <a class="govuk-link" href="https://www.gov.uk" target="_blank" rel="noopener noreferrer">GOV.UK (opens in new tab)</a> and search
+          <a class="govuk-link" href="https://www.gov.uk/government/organisations/rural-payments-agency/about/personal-information-charter" target="_blank" rel="noopener noreferrer">Rural Payments Agency personal information charter (opens in new tab).</a>
+        </p>
+      </div>
+    </div>
+  {% endif %}
+
+  <script>
```
Comment (swdpcomputing) [inline src/views/pages/consent-confirmation.njk]: A lot of this is going to be reusable. Shall we abstract it out?

**Comment by swdpcomputing** [comment]
We have some missing coverage (not sure whether that's due to this PR) - 
image

----
UPDATE - this has been improved now

File: src/config/machines/example-grant-machine.js (hunk 1 above)
Comment (Andrew-Folga) [reply src/config/machines/example-grant-machine.js]: Let's talk on a huddle.

File: src/views/pages/consent-confirmation.njk (hunk 2 above)
Comment (Andrew-Folga) [reply src/views/pages/consent-confirmation.njk]: Let's talk on a huddle.

File: src/config/machines/example-grant-machine.js (hunk 1 above)
Comment (Andrew-Folga) [reply src/config/machines/example-grant-machine.js]: Applied

File: src/views/pages/consent-confirmation.njk (hunk 2 above)
Comment (Andrew-Folga) [reply src/views/pages/consent-confirmation.njk]: Done

**Review by ScenarioFactory — APPROVED** [review]

**Review by swdpcomputing — APPROVED** [review]

**Pull request merged**